### ISSUE TITLE
TableReporterVector not adding all channels to tabel

### ISSUE THREAD

**Issue by clnsmith** [opensim-org/opensim-core#1792]
Not sure if this is an implementation error or a user error:

Right now I am doing something which boils down to this:

TableReporterVector reporter = new TableReporterVector();
reporter->setName("vec_report");

reporter->addToReport(frc.getOutput("tri_data"));  //SimTK::Vector Output
reporter->addToReport(frc.getOutput("vertex_data"));   //SimTK::Vector Output

model.addComponent(reporter);

run simulaton...

TableReporterVector rep = model.getComponent<TableReporterVector>("vec_report");

std::vector<std::string> labels = rep.getTable().getColumnLabels();

but here labels only contains "tri_data[0]" ... "tri_data[end]". There is no data for the "vertex_data" output I added to the reporter.

**Comment by chrisdembia** [comment]
Hmm...loooking at the code (https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/Reporter.h), I would have expected that the labels are `{"tri_data", "vertex_data"}`. Are the `tri_data` columns correct?

**Comment by aseth1** [comment]
I believe `TableReporter` is specialized for `Vector<double>` to expect and report only one input (see [here](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/Reporter.h#L331). A list input of Vectors would require looping over the number inputs as is in the generic implementation.

**Comment by clnsmith** [comment]
hmm I made the simple case I wrote out above, and the column labels come out as Chris described. 

Within my more complex code I am still getting the column labels as I described above. I'm trying to isolate what the difference between the two cases are, i'll keep you posted.

**Comment by clnsmith** [comment]
So heres the difference:

If I run a simulation (ie manager.integrate) then the labels work out as Chris suggests. 

If don't run a simulation, and just pose the model (ie model.realizeReport(state)) as I do in an analysis, then I get the case Ajay pointed out.

**Comment by chrisdembia** [comment]
@clnsmith thanks for the additional information. What about the data themselves? Do they correspond with the column labels?

**Comment by chrisdembia** [comment]
There might be an issue with the fact that column labels are set in two different places ([here](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/Reporter.h#L231) and [here](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/Reporter.h#L339)), but the code for actually appending data seems to be the same in both cases.

**Comment by aseth1** [comment]
You're right. The specialized `implementReport` should not overwrite the labels, instead it could create additional column meta data that are the element-labels (for elements of the Vector) instead of hijacking the labels obtained from the inputs.

**Comment by chrisdembia** [comment]
@clnsmith by default, the Manager does not cause any reporting with the new Reporters. You must set a `report_time_interval`. Can you check if the table from the manager has any rows?

**Comment by clnsmith** [comment]
Good call. If I set a report_time_interval then the manager.integrate gives the single input (tri_data[0]... ) result. The data looks correct for that single input, (although it is Vector with 8000 elements, so I could be missing something). 

I think just adding documentation that the Vector reporter can only take one input is enough to prevent my error. In the long term it would be nice to be able to use multiple inputs (either generate a table for each input or flatten them all into one table).
